src/arithmetic.jl

```diff
@@ -489,8 +493,9 @@ term of the denominator.
         return nothing
     end
 
-    @inbounds for i = 0:k-1
-        k+ordfact-i > b.order && continue
+    imin = max(0, k+ordfact-b.order)
+    @inbounds c[k] = c[imin] * b[k+ordfact-imin]
+    @inbounds for i = imin+1:k-1
```

Comment:
Very nice!

Reply:
Thanks! I actually took inspiration from `pow!`, where similar things were already implemented to optimize a similar loop